Description: set correct dynamic linker path for armhf
 armhf uses a different dynamic linker path from armel
Author: Peter Michael Green <plugwash@debian.org>
Last-Update: 2012-11-14
Reviewed-By: Michael Stapelberg <stapelberg@debian.org>
Bug-Debian: http://bugs.debian.org/693208
Forwarded: not-needed

---

Index: clang-3.0/tools/clang/lib/Driver/Tools.cpp
===================================================================
--- clang-3.0.orig/tools/clang/lib/Driver/Tools.cpp	2012-11-13 12:04:59.000000000 +0000
+++ clang-3.0/tools/clang/lib/Driver/Tools.cpp	2012-11-13 12:04:59.000000000 +0000
@@ -4315,7 +4315,10 @@
       CmdArgs.push_back("/lib/ld-linux.so.2");
     else if (ToolChain.getArch() == llvm::Triple::arm ||
              ToolChain.getArch() == llvm::Triple::thumb)
-      CmdArgs.push_back("/lib/ld-linux.so.3");
+      if (ToolChain.getTriple().getEnvironment() == llvm::Triple::GNUEABIHF)
+        CmdArgs.push_back("/lib/ld-linux-armhf.so.3");
+      else
+        CmdArgs.push_back("/lib/ld-linux.so.3");
     else if (ToolChain.getArch() == llvm::Triple::ppc)
       CmdArgs.push_back("/lib/ld.so.1");
     else if (ToolChain.getArch() == llvm::Triple::ppc64)
